### DISCUSSION TITLE
How can i update a custom SSL certificate?

### DISCUSSION THREAD

**Discussion by uebmaster** [EasyEngine/easyengine#1697] [Support Requests] [answered]
Hi.

My custom ssl certificate is soon to expire, and i want to update it in my ee sites, but i don't find a solution to do it.

I've tried in a test site (it has let's encrypt enabled) with:

`sudo ee site update example.com --ssl=off`

and later i've tried:

`sudo ee site update example.com --ssl=custom  --ssl-key='/path/to/certificate.key' --ssl-crt='/path/to/certificate.crt'`

But i got:

```
Error: Parameter errors:
 unknown --ssl-key parameter
 unknown --ssl-crt parameter
```

Then the site redirect me to another site that i have in the same server.

I don't know what to do.

I would appreciate if someone could help me.

Thanks in advance.

```
+-------------------+-----------------------------------------------------------------------------+
| OS                | Linux 4.15.0-163-generic #171-Ubuntu SMP Fri Nov 5 11:55:11 UTC 2021 x86_64 |
| Shell             | /bin/bash                                                                   |
| PHP binary        | /usr/bin/php7.2                                                             |
| PHP version       | 7.2.34-28+ubuntu18.04.1+deb.sury.org+1                                      |
| php.ini used      | /etc/php/7.2/cli/php.ini                                                    |
| EE root dir       | phar://ee.phar                                                              |
| EE vendor dir     | phar://ee.phar/vendor                                                       |
| EE phar path      | /home/uebmaster                                                             |
| EE packages dir   |                                                                             |
| EE global config  | /opt/easyengine/config/config.yml                                           |
| EE project config |                                                                             |
| EE version        | 4.5.5                                                                       |
+-------------------+-----------------------------------------------------------------------------+
```
```
NAME="Ubuntu"
VERSION="18.04.6 LTS (Bionic Beaver)"
```

**Comment by dhsathiya** [comment, marked as answer]
Seems like the `site update` command doesn't support custom certificate updates. 

For now, let's take a little longer approach for this

0. Before starting with it, read the whole answer :)
1. Take a backup of `/opt/easyengine/services/nginx-proxy/certs/` and all the certificates inside it. (check the permissions as well)
2. Make sure the backup destination contains all the certs.
3. Run `ls` in the directory and you will find the `.crt` and `.key` files there - `example.com.crt` and `example.com.key`
4. Use vim/your favorite editor to update the contents of the `.key` and `.crt` file otherwise you can replace the files with the new certs making sure the file name remains the same.
5. Run `ee service restart nginx-proxy`

Want to be double sure?
- Take a backup (Snapshot/AMI) of the whole machine and then proceed with the below steps.

What if it fails?
- Your site(s) will go down. Until you revert your changes from the backup.
   - you just have to get the old certificates back to the `/opt/easyengine/services/nginx-proxy/certs/` directory and run `ee service restart nginx-proxy`

**Reply by Chipmank** [reply]
I have another issue. The site was created with the self-signed certificate, then I've installed a certificate from sectigo but now I'm getting errors with "certificate chain is incomplete". How to replace the original cert with the proper one?

**Reply by iShariefi** [reply]
[This tool](https://whatsmychaincert.com) should solve your issue...
